src/main/kotlin/com/milaboratory/mixcr/cli/CommandBuildLibrary.kt

```diff
@@ -0,0 +1,394 @@
+                field = value
+            }
+    }
+
+    @ArgGroup(exclusive = true, multiplicity = "0..1")
+    var cGenes: CGenesInput? = null
+
+    class CGenesInput : GenesInput {
+        @ArgGroup(exclusive = false)
+        override var fasta: CGenesFasta? = null
+
+        class CGenesFasta : GenesFasta {
+            @set:Option(
+                names = ["--c-genes-from-fasta"],
+                description = ["FASTA file with Constant genes"],
+                paramLabel = "<j.fasta>",
+                required = true
+            )
+            override var pathOps: Path? = null
+                set(value) {
+                    ValidationException.requireFileType(value, InputFileType.FASTA)
+                    field = value
+                }
+
+            @set:Option(
+                names = ["--c-gene-feature"],
+                description = ["Gene feature corresponding to Constant gene sequences in FASTA (CExon1 by default)"],
+                paramLabel = "<gene_feature>",
+                required = false
+            )
+            override var geneFeatureOps: String? = "CExon1"
+                set(value) {
+                    ValidationException.requireKnownGeneFeature(value)
+                    field = value
+                }
+        }
+
+        @set:Option(
+            names = ["--c-genes-from-species"],
+            description = ["Species to take Constant genes from it (human, mmu, lamaGlama, alpaca, rat, spalax"],
+            paramLabel = "<species>",
+            required = true
+        )
+        override var species: String? = null
+            set(value) {
+                ValidationException.requireKnownSpecies(value)
+                field = value
+            }
+    }
+
+    @set:Option(
+        names = ["--species"],
+        description = ["Species names"],
+        paramLabel = "<species name>",
+        required = true,
+
+        )
+    var species: String = ""
+
+    @set:Option(
+        names = ["--taxon-id"],
+        description = ["Taxon ID"],
+        paramLabel = "<taxon_id>",
+    )
+    var taxonId: String = "0"
+
+    @set:Option(
+        names = ["--chain"],
+        description = ["Immunological chain"],
+        paramLabel = "<chain>",
+        required = true
+    )
+    var chain: String = ""
+        set(value) {
+            ValidationException.requireKnownChains(value)
+            field = value
+        }
+
+    @set:Option(
+        names = ["--do-not-infer-points"],
+        description = ["Do not infer reference points"],
+        required = false
+    )
+    var doNotInferPoints: Boolean = false
+
+    @set:Option(
+        names = ["--keep-intermediate"],
+        description = ["Keep intermediate files"],
+        required = false
+    )
+    var keepIntermediateFiles: Boolean = false
+
+    @Parameters(
+        description = ["Output library."],
+        paramLabel = "library.json",
+        index = "0",
+    )
+    var output: Path? = null
```

Comment:
it's not optional by your code

Reply:
Yes. But you can't declare var without value and you can't use lateinit with custom setter

Reply:
done